utils/fragment_range.hh

```diff
@@ -173,3 +173,28 @@ concept FragmentedMutableView = requires (T view) {
     requires std::is_same_v<typename T::fragment_type, bytes_mutable_view>;
 };
 
+template<FragmentedView View>
+requires (!FragmentRange<View>)
+bytes linearized(View v)
+{
+    bytes b(bytes::initialized_later(), v.size_bytes());
+    auto out = b.begin();
+    while (v.size_bytes()) {
+        out = std::copy(v.current_fragment().begin(), v.current_fragment().end(), out);
+        v.remove_current();
+    }
+    return b;
+}
+
+template<FragmentedView View, typename Function>
+requires (!FragmentRange<View>) && requires (Function fn, bytes_view bv) {
+    fn(bv);
+}
```

Comment:
You can write the latter constraint as std::invocable<Function, bytes_view>

Reply:
Done.